modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/ConcatWorkflowOperationHandler.java

```diff
@@ -192,39 +209,42 @@ private WorkflowOperationResult concat(MediaPackage src, WorkflowOperationInstan
 
-    // Output resolution
-    if (outputResolution == null)
-      throw new WorkflowOperationException("Output resolution must be set!");
+    // Output resolution - if not keeping dimensions the same, it must be set
+    if (!sameCodec && outputResolution == null)
+        throw new WorkflowOperationException("Output resolution must be set!");
 
     Dimension outputDimension = null;
-    if (outputResolution.startsWith(OUTPUT_PART_PREFIX)) {
-      if (!trackSelectors.keySet().contains(
-              Integer.parseInt(outputResolution.substring(OUTPUT_PART_PREFIX.length()))))
-        throw new WorkflowOperationException("Output resolution part not set!");
-    } else {
-      try {
-        String[] outputResolutionArray = StringUtils.split(outputResolution, "x");
-        if (outputResolutionArray.length != 2) {
-          throw new WorkflowOperationException("Invalid format of output resolution!");
+    if (!sameCodec) { // Ignore resolution if same Codec - no scaling
+      if (outputResolution.startsWith(OUTPUT_PART_PREFIX)) {
+        if (!trackSelectors.keySet().contains(
+            Integer.parseInt(outputResolution.substring(OUTPUT_PART_PREFIX.length()))))
+          throw new WorkflowOperationException("Output resolution part not set!");
+      } else {
+        try {
+          String[] outputResolutionArray = StringUtils.split(outputResolution, "x");
+          if (outputResolutionArray.length != 2) {
+            throw new WorkflowOperationException("Invalid format of output resolution!");
+          }
+          outputDimension = Dimension.dimension(Integer.parseInt(outputResolutionArray[0]),
+            Integer.parseInt(outputResolutionArray[1]));
+        } catch (WorkflowOperationException e) {
+          throw e;
+        } catch (Exception e) {
+          throw new WorkflowOperationException("Unable to parse output resolution!", e);
         }
-        outputDimension = Dimension.dimension(Integer.parseInt(outputResolutionArray[0]),
-                Integer.parseInt(outputResolutionArray[1]));
-      } catch (WorkflowOperationException e) {
-        throw e;
-      } catch (Exception e) {
-        throw new WorkflowOperationException("Unable to parse output resolution!", e);
       }
     }
 
     float fps = -1.0f;
-    if (StringUtils.isNotEmpty(outputFrameRate)) {
+    // Ignore fps if same Codec - no scaling
+    if (!sameCodec && StringUtils.isNotEmpty(outputFrameRate)) {
       if (StringUtils.startsWith(outputFrameRate, OUTPUT_PART_PREFIX)) {
-        if (!NumberUtils.isNumber(outputFrameRate.substring(OUTPUT_PART_PREFIX.length()))
+        if (!NumberUtils.isCreatable(outputFrameRate.substring(OUTPUT_PART_PREFIX.length()))
                 || !trackSelectors.keySet().contains(Integer.parseInt(
                         outputFrameRate.substring(OUTPUT_PART_PREFIX.length())))) {
           throw new WorkflowOperationException("Output frame rate part not set or invalid!");
```

Comment:
It would be helpful to know which one is bad.  Also, what's the format here?  I'm guessing something like part-n-25, which isn't going to format correctly with the given code.  This format does not appear to be in the docs as far as I can see.